### ISSUE TITLE
vscode neovim extension's forward search triggers strange UI behavior

### ISSUE THREAD

**Issue by kting28** [cdr/code-server#4294]
## OS/Web Information

- Web Browser: Chrome  93.0.4577.82
- Local OS: Windows 10
- Remote OS: Linux
- Remote Architecture: amd64
- `code-server --version`: 3.12.0 4cd55f94c0a72f05c18cea070e10b969996614d2

## Steps to Reproduce

1. Install VScode Neovim extension https://github.com/asvetliakov/vscode-neovim
2. Configure extension with path to nvim
2. open any text file
3. press '/'  and rapidly start typing characters to search.

## Expected

vim forward search should happen
## Actual
The forward search text cursor jumps back and forth rapidly and eventually stops. Any fast text entry at this particular prompt seems to cause the issue.

## Logs
I can provide logs but it seems when this behavior happens, there's no new messages printed either on browser console nor code-server. If you still need them let me know.

## Screenshot
![code-server-nvim-cursor](https://user-images.githubusercontent.com/24449116/135699650-c41c4fc4-96fb-4cfe-87d6-22e97c5a2bbf.gif)

## Notes

This issue can be reproduced in VS Code: No

**Comment by jsjoeio** [comment]
Thanks for the notes! I was able to reproduce. I don't see any logs though so I'm not sure what's going on.

Can you reproduce this in Codespaces? (If not, then this must be an issue with code-server)

## Video


https://user-images.githubusercontent.com/3806031/135894597-aeee9057-6124-41a7-bd58-a75347305877.mov

**Comment by kting28** [comment]
Thanks for looking into this. I think the only codespaces access I have is the vscode that launches on github.com with the "." key. From there it seems neovim is just not supporte
<img width="615" alt="Capture" src="https://user-images.githubusercontent.com/24449116/135940603-ca7af152-3ecd-4ad2-b1d2-a773fe411716.PNG">
d

**Comment by jsjoeio** [comment]
> I think the only codespaces access I have is the vscode that launches on github.com with the "." key. From there it seems neovim is just not supported

Aw bummer. Yeah, github.dev is client only and it appears neovim relies on neovim being on the underlying server so makes sense it's not supported there.

Hmm...I took a look with Codespaces and got this error:

![image](https://user-images.githubusercontent.com/3806031/136081750-aeda8c48-b785-422d-863f-8e379d326be5.png)

> Examples are when an extension accesses files from the VS Code installation on the desktop or when an extension depends on an executable that must run in a desktop environment. When you try to install such an extension in the browser, you will be informed that the extension is not available.

It sounds like it's not supported in the web 🤔

In that case, I'm not sure how much we can help here since it's not even supported upstream. I don't even know how I'd go about troubleshooting this.

Maybe you could file an upstream issue?

Also maybe [try this](https://github.com/asvetliakov/vscode-neovim/issues/683#issuecomment-917229225)/

**Comment by kting28** [comment]
Yup no problem, in the grand scheme of things this is minor. I will close the ticket for now and just leave this as a record. May be if we find other UI operations that trigger the same behavior we will have more clues. Thanks a bunch.
